bower_components/ng-token-auth/src/ng-token-auth.coffee

```diff
@@ -0,0 +1,903 @@
+      newHeaders = {}
+      for key, val of $auth.getConfig().tokenFormat
+        if resp.headers(key)
+          newHeaders[key] = resp.headers(key)
+
+      if tokenIsCurrent($auth, newHeaders)
+        $auth.setAuthHeaders(newHeaders)
+
+    # this is ugly...
+    # we need to configure an interceptor (must be done in the configuration
+    # phase), but we need access to the $http service, which is only available
+    # during the run phase. the following technique was taken from this
+    # stackoverflow post:
+    # http://stackoverflow.com/questions/14681654/i-need-two-instances-of-angularjs-http-service-or-what
+    $httpProvider.interceptors.push ['$injector', ($injector) ->
+      request: (req) ->
+        $injector.invoke ['$http', '$auth',  ($http, $auth) ->
+          if req.url.match($auth.apiUrl())
+            for key, val of $auth.retrieveData('auth_headers')
+              req.headers[key] = val
+        ]
+
+        return req
+
+      response: (resp) ->
+        $injector.invoke ['$http', '$auth', ($http, $auth) ->
+          if resp.config.url.match($auth.apiUrl())
+            return updateHeadersFromResponse($auth, resp)
+        ]
+
+        return resp
+
+      responseError: (resp) ->
+        $injector.invoke ['$http', '$auth', ($http, $auth) ->
+          if resp.config.url.match($auth.apiUrl())
+            return updateHeadersFromResponse($auth, resp)
+        ]
+
+        return $injector.get('$q').reject(resp)
+    ]
+
+    # define http methods that may need to carry auth headers
+    httpMethods = ['get', 'post', 'put', 'patch', 'delete']
+
+    # disable IE ajax request caching for each of the necessary http methods
+    angular.forEach(httpMethods, (method) ->
+      $httpProvider.defaults.headers[method] ?= {}
+      $httpProvider.defaults.headers[method]['If-Modified-Since'] = 'Mon, 26 Jul 1997 05:00:00 GMT'
+    )
+  ])
+
+  .run(['$auth', '$window', '$rootScope', ($auth, $window, $rootScope) ->
+    $auth.initialize()
+  ])
+
+# ie8 and ie9 require special handling
+window.isOldIE = ->
+  out = false
+  nav = navigator.userAgent.toLowerCase()
+  if nav and nav.indexOf('msie') != -1
+    version = parseInt(nav.split('msie')[1])
+    if version < 10
+      out = true
+
+  out
+
+# ie <= 11 do not support postMessage
+window.isIE = ->
+  nav = navigator.userAgent.toLowerCase()
+  ((nav and nav.indexOf('msie') != -1) || !!navigator.userAgent.match(/Trident.*rv\:11\./))
+
+window.isEmpty = (obj) ->
+  # null and undefined are "empty"
+  return true unless obj
+
+  # Assume if it has a length property with a non-zero value
+  # that that property is correct.
+  return false if (obj.length > 0)
+  return true if (obj.length == 0)
+
+  # Otherwise, does it have any properties of its own?
+  # Note that this doesn't handle
+  # toString and valueOf enumeration bugs in IE < 9
+  for key, val of obj
+    return false if (Object.prototype.hasOwnProperty.call(obj, key))
+
+  return true
```

Comment:
yep, this was intended to be quick and dirty proof of concept. will re-do/re-commit all this work to clean that up.